CMakeLists.txt

```diff
@@ -343,21 +344,31 @@ if(MZ_ZSTD)
         find_package(PkgConfig QUIET)
         if(PKGCONFIG_FOUND)
             pkg_check_modules(ZSTD libzstd)
+            if(ZSTD_FOUND)
+                list(APPEND MINIZIP_INC ${ZSTD_INCLUDE_DIRS})
+                list(APPEND MINIZIP_LIB ${ZSTD_LIBRARIES})
+                list(APPEND MINIZIP_LBD ${ZSTD_LIBRARY_DIRS})
+                set(PC_PRIVATE_LIBS "${PC_PRIVATE_LIBS} -lzstd")
+                list(APPEND MINIZIP_DEP_PKG ZSTD)
+            endif()
         endif()
         if(NOT ZSTD_FOUND)
             find_package(ZSTD QUIET)
-            set(ZSTD_VERSION ${ZSTD_VERSION_STRING})
+            if(ZSTD_FOUND)
+                set(ZSTD_VERSION ${ZSTD_VERSION_STRING})
+                if(TARGET libzstd_static)
```

Comment:
Perhaps this should be:
```
if(NOT DEFINED BUILD_SHARED_LIBS OR NOT ${BUILD_SHARED_LIBS})
```
Like on line 379.

Reply:
because there is not always the case that we would get shared libs, for e.g. if vcpkg is used to link zlib in Linux, then build would fail, because vcpkg don't support shared in linux